app/controllers/sign_up/completions_controller.rb

```diff
@@ -99,8 +99,7 @@ def track_completion_event(last_page)
     end
 
     def pii
-      pii_string = Pii::Cacher.new(current_user, user_session).fetch_string
-      JSON.parse(pii_string || '{}', symbolize_names: true)
+      Pii::Cacher.new(current_user, user_session).fetch&.to_h || {}
```

Comment:
I think this will mostly "just work" to keep it as an Attributes object rather than a hash

```suggestion
      Pii::Cacher.new(current_user, user_session).fetch || Pii::Attributes.new
```

Reply:
If it does, should also switch to using Pii::Attributes in the spec

```diff
--- a/spec/presenters/completions_presenter_spec.rb
+++ b/spec/presenters/completions_presenter_spec.rb
@@ -13,18 +13,20 @@ RSpec.describe CompletionsPresenter do
   let(:current_user) { create(:user, :fully_registered, identities: identities) }
   let(:current_sp) { create(:service_provider, friendly_name: 'Friendly service provider') }
   let(:decrypted_pii) do
-    {
-      first_name: 'Testy',
-      last_name: 'Testerson',
-      ssn: '900123456',
-      address1: '123 main st',
-      address2: 'apt 123',
-      city: 'Washington',
-      state: 'DC',
-      zipcode: '20405',
-      dob: '1990-01-01',
-      phone: '+12022121000',
-    }
+    Pii::Attributes.new_from_hash(
+      {
+        first_name: 'Testy',
+        last_name: 'Testerson',
+        ssn: '900123456',
+        address1: '123 main st',
+        address2: 'apt 123',
+        city: 'Washington',
+        state: 'DC',
+        zipcode: '20405',
+        dob: '1990-01-01',
+        phone: '+12022121000',
+      }
+    )
```

Reply:
It does in fact just work. I just pushed a commit with some little changes.